src/controllers/treatment.controller.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import TreatmentController from './treatment.controller';
+import { TreatmentService } from '../services';
+import { CreateTreatmentDTO, UpdateTreatmentDTO } from '../dtos';
+
+describe('TreatmentController', () => {
+  let controller: TreatmentController;
+  let service: TreatmentService;
+
+  const mockTreatmentService = {
```

Comment:
I see that you're mocking the functions of the controller to execute your specs, this is not ideal because you're not really testing the logic in your controllers but instead you're pre-defining the responses of the methods without going through the logic in it.

I'd suggest to remove these mocks and instead create real db records with typeorm API  to execute the specs so you can actually test your controller logic